concent_api/core/payments/backends/sci_backend.py

```diff
@@ -58,25 +58,25 @@ def get_list_of_payments(
     return payments_list
 
 
-def validate_that_all_payment_ts_are_younger_than_last_payment_closure_time_if_payment_exists(
+def validate_that_last_closure_time_is_older_than_last_payment(
     requestor_eth_address: str,
     provider_eth_address: str,
-    oldest_payment_ts: Optional[int],
+    youngest_payment_ts: Optional[int],
 ) -> None:
-    if oldest_payment_ts is None:
+    if youngest_payment_ts is None:
         return
     forced_payment_event_list = _get_list_of_forced_payment_events(
         requestor_eth_address=requestor_eth_address,
         provider_eth_address=provider_eth_address,
-        search_payments_since_ts=oldest_payment_ts,
+        search_payments_since_ts=youngest_payment_ts,
     )
     for forced_payment_event in forced_payment_event_list:
-        if oldest_payment_ts < forced_payment_event.closure_time:
+        if youngest_payment_ts < forced_payment_event.closure_time:
```

Comment:
again, the function says something and than does something else: you are not comparing `last_payment` (`youngest_payment_ts`) with **last** `closure_time`, by with a **list** of closure times... Please change the implementation

Reply:
renamed, is is ok now?

Reply:
quite frankly I would prefer having the previous name and the code changed:
why not just
```python
if youngest_payment_ts < max([event.closure_time for events in forced_payment_event_list]):
    raise BanksterTimestampError
```
instead of the `for` loop? This way name of the function stays more meaningful and the implementation better reflects the nature of a problem.

Reply:
ok, done